projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -508,990 +324,646 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
-            takeUntil(this._destroy$)).subscribe((eventArgs) => {
-                this._onOpening(eventArgs);
-            });
-
-        this._overlayService.onOpened.pipe(
-            filter((overlay) => overlay.id === this._componentID),
-            takeUntil(this._destroy$)).subscribe(() => {
-                this._onOpened();
-            });
-
-        this._overlayService.onClosed.pipe(
-            filter(overlay => overlay.id === this._componentID),
-            takeUntil(this._destroy$)).subscribe(() => {
-                this._onClosed();
-            });
-
-        this._overlayService.onClosing.pipe(
-            filter(overlay => overlay.id === this._componentID),
-            takeUntil(this._destroy$)).subscribe((event) => {
-                this.onClosing.emit(event);
-                // If canceled in a user onClosing handler
-                if (event.cancel) {
-                    return;
+    @HostListener('keydown', ['$event'])
+    public onKeyDown(event: KeyboardEvent) {
+        switch (event.key) {
+            case KEYS.UP_ARROW:
+            case KEYS.UP_ARROW_IE:
+                if (event.altKey) {
+                    this.close();
                 }
-                // Do not focus the input if clicking outside in dropdown mode
-                const input = this.getEditElement();
-                if (input && !(event.event && this.mode === InteractionMode.DropDown)) {
-                    input.focus();
-                } else {
-                    // outside click
-                    this._updateValidityOnBlur();
+                break;
+            case KEYS.DOWN_ARROW:
+            case KEYS.DOWN_ARROW_IE:
+                if (event.altKey) {
+                    this.open();
                 }
-            });
-
-        if (this.mode === InteractionMode.DropDown) {
-            this.dateFormatParts = DatePickerUtil.parseDateFormat(this.mask, this.locale);
-            if (this.mask === undefined) {
-                this.mask = DatePickerUtil.getMask(this.dateFormatParts);
-            }
-            this.inputMask = DatePickerUtil.getInputMask(this.dateFormatParts);
+                break;
+            case KEYS.ESCAPE:
+            case KEYS.ESCAPE_IE:
+                this.close();
+                break;
         }
-
-        this._ngControl = this._injector.get<NgControl>(NgControl, null);
     }
 
-    /** @hidden @internal */
-    public ngAfterViewInit() {
-        if (this.mode === InteractionMode.DropDown && this._inputElementRef) {
-            fromEvent(this._inputElementRef.nativeElement, 'keydown').pipe(
-                throttle(() => interval(0, animationFrameScheduler)),
-                takeUntil(this._destroy$)
-            ).subscribe((res) => this.onKeyDown(res));
-        }
+    /**
+     * Selects today's date.
+     *
+     * @example
+     * ```typescript
+     * this.datePicker.selectToday();
+     * ```
+     */
+    public selectToday(): void {
+        this.select(new Date());
```

Comment:
Did we talk about removing this? Technically it should be new Date with cleared time portion, but that is prone to shift and I really would rather not go there if we can :)

Reply:
In the end, there's some value to this method so we're keeping it after all